uwblueprint/focus-on-nature
# feat: create get camp endpoint

## Notion ticket link

[Get Camps](https://www.notion.so/uwblueprintexecs/Tasks-38b49f83f6ce4b07ae188dabb37631e8?p=e8830a0c18d04bb7bebf1a8c23397c4f)



## Implementation description
* fixed baseCamp type in camp.model.ts from `Schema.Types.ObjectId` to `BaseCamp | Schema.Types.ObjectId;` in order to support mongoose populate
* added a GET camp route - /camp
* added getCamps service that returns an array of GetCampDTO objects
* fixed camp csv service


## Steps to test
1. when there are no camps, the API should return an empty array []
2. you can call POST /camp/ to create a camp, and then GET /camp/ to view the camp

Call POST /camp with the following JSON body to create a camp:
`{
    "active": true,
    "ageLower": 15,
    "ageUpper": 30,
    "name": "test camp",
    "description": "description",
    "location": "canada",
    "capacity": 20,
    "startTime": "12:00",
    "endTime": "17:00",
    "dates": ["Sun Mar 13 2022 20:01:14 GMT-0600 (Mountain Daylight Time)"],
    "fee": 25,
    "formQuestions": [{"question": "Q1", "type": "Text", "required": true, "description": "question description"}]
}`

Call GET /camp to get all the camps

Sample Response:

![Screenshot 2022-03-13 at 8 26 54 PM](https://user-images.githubusercontent.com/67038446/158094432-68d23333-2f76-4306-92be-ba4aa46bab55.png)
9250baf6fceb.png)



## What should reviewers focus on?
* verify if "get camp" route response contain all the desired fields

## Checklist
